src/Umbraco.Web.UI.Client/src/views/propertyeditors/datepicker/datepicker.controller.js

```diff
@@ -78,9 +78,20 @@ function dateTimePickerController($scope, notificationsService, assetsService, a
         setDatePickerVal();
     };
 
-    $scope.inputChanged = function() {
-        setDate($scope.model.datetimePickerValue);
-        setDatePickerVal();
+    $scope.inputChanged = function () {        
+        if ($scope.model.datetimePickerValue == "" && $scope.hasDatetimePickerValue) {
+                $scope.clearDate();
+        } else if ($scope.model.datetimePickerValue) {
+            var momentDate = moment($scope.model.datetimePickerValue, $scope.model.config.format, true);
+            if (!momentDate || !momentDate.isValid()) {
+                momentDate = moment(new Date($scope.model.datetimePickerValue));
```

Comment:
For my understanding: this happens when the first time parsing the date with the current configuration fails, correct? When would we get `!momentDate` and when `!momentDate.isValid()`?